lib/ansible/modules/cloud/amazon/lambda.py

```diff
@@ -430,6 +432,9 @@ def main():
 
     # Function doesn't exists, create new Lambda function
     elif state == 'present':
+        if not role_arn:
+            module.fail_json(msg='A role is required for Lambda function creation')
```

Comment:
This should probably be done with `required_if=[['state', 'present', ['role_arn']]]` in the `argument_spec`